ruby_event_store/lib/ruby_event_store/client.rb

```diff
@@ -171,6 +171,16 @@ def within(&block)
       Within.new(block, event_broker)
     end
 
+    def with_metadata(metadata, &block)
+      previous_metadata = self.metadata
+      begin
```

Comment:
i think we could use just `ensure` without `begin`

Reply:
yes, you're right